enclave-tls/src/lib/tls_wrappers/null/main.c

```diff
@@ -0,0 +1,43 @@
+#include <enclave-tls/tls_wrapper.h>
+#include <enclave-tls/err.h>
+#include <enclave-tls/log.h>
+
+extern tls_wrapper_err_t tls_wrapper_register(tls_wrapper_opts_t *);
+extern tls_wrapper_err_t null_pre_init(void);
+extern tls_wrapper_err_t null_init(tls_wrapper_ctx_t *);
+extern tls_wrapper_err_t null_gen_pubkey_hash(tls_wrapper_ctx_t *,
+					      enclave_tls_cert_algo_t,
+					      uint8_t *);
+extern tls_wrapper_err_t null_gen_cert(tls_wrapper_ctx_t *,
+				       const tls_wrapper_cert_info_t *);
+extern tls_wrapper_err_t null_negotiate(tls_wrapper_ctx_t *, int fd);
+extern tls_wrapper_err_t null_transmit(tls_wrapper_ctx_t *, void *, size_t *);
+extern tls_wrapper_err_t null_receive(tls_wrapper_ctx_t *, void *, size_t *);
+extern tls_wrapper_err_t null_cleanup(tls_wrapper_ctx_t *);
+
+static tls_wrapper_opts_t opts_test = {
+	.version = TLS_WRAPPER_API_VERSION_DEFAULT,
+	.type = "null",
+	.priority = 0,
+	.pre_init = null_pre_init,
+	.init = null_init,
+	.gen_pubkey_hash = null_gen_pubkey_hash,
+	.gen_cert = null_gen_cert,
+	.negotiate = null_negotiate,
+	.transmit = null_transmit,
+	.receive = null_receive,
+	.cleanup = null_cleanup,
+};
+
+/* *INDENT-OFF* */
+void __attribute__((constructor))
+libtls_wrapper_null_init(void)
+{
+	ETLS_DEBUG("The constructor of libtls_wrapper_null.so is called\n");
+
+	tls_wrapper_err_t err = tls_wrapper_register(&opts_test);
+	if (err != TLS_WRAPPER_ERR_NONE) {
+		ETLS_ERR("ERROR: failed to tls_wrapper_register()\n");
```

Comment:
I think all error messages should be more readable than debug info right? So this error message should be "failed to register tls wrapper instance \"NULL\"".